### PR TITLE
themedScript Added

### PR DESCRIPTION
**Pull request by tony13tv**
A function to call javascript files just by its name as it occurs with themedCSS

### CODE REVIEW

**Comment by sminnee** [comment]
It's a reasonable addition to the api. For consistency my preference would be to call it `themedJavascript()`.

Also, ideally we'd have some tests for this, although I note that we don't yet have tests for `themedCSS()` either. Are you familiar with creating tests, @tony13tv? It would be awesome if you could get a basic test for both `themedCSS()` and `themedJavascript()`.

File: view/Requirements.php [hunk 1]
```diff
@@ -1332,6 +1349,38 @@ public function themedCSS($name, $module = null, $media = null) {
...
+		$absproject = $absbase . $project;
+		$js = "/javascript/$name.js";
+
+		if(file_exists($absproject . $js)) {
+			$this->javascript($project . $js);
+		} elseif($module && file_exists($abstheme . '_' . $module.$js)) {
+			$this->javascript($theme . '_' . $module . $js);
+		} elseif(file_exists($abstheme . $js)) {
+			$this->javascript($theme . $js);
+		} elseif($module) {
```
Comment (tractorcow) [inline view/Requirements.php]: I would prefer a check that the file exists here, and throw an Exception if the file cannot be loaded in any context.

**Comment by tractorcow** [comment]
This needs unit tests please. :)

**Comment by tractorcow** [comment]
If you need to mock BASE_PATH for testing you can use Director::baseFolder(), and then setup a 'testing site', as other tests do.

File: view/Requirements.php (hunk 1 above)
Comment (tony13tv) [inline view/Requirements.php]: That is a great addition! I will make the changes in both themedCSS and themedJavascript and will throw the exception only if the site is in dev mode.

**Comment by tractorcow** [comment]
Thanks for the feedback on the error handling. How do you feel about having tests? :D

File: view/Requirements.php (hunk 1 above)
Comment (tractorcow) [inline view/Requirements.php]: If it's an error, then it should be thrown in all contexts, not only in dev mode.

File: view/Requirements.php [hunk 2]
```diff
@@ -1328,6 +1345,42 @@ public function themedCSS($name, $module = null, $media = null) {
 			$this->css($theme . $css, $media);
 		} elseif($module) {
 			$this->css($module . $css, $media);
+		} elseif (Director::isDev()) {
+			throw Exception("The css file doesn't exists. Please check if the file $name.css exists in any context or search for themedCSS references calling this file in your templates.");
```
Comment (dhensby) [inline view/Requirements.php]: I think an `InvalidArgumentException` makes most sense here.

**Comment by dhensby** [comment]
I'm going to assume that if we want test coverage, we need to write it ourselves

File: view/Requirements.php [hunk 3]
```diff
@@ -1328,6 +1345,42 @@ public function themedCSS($name, $module = null, $media = null) {
 			$this->css($theme . $css, $media);
 		} elseif($module) {
 			$this->css($module . $css, $media);
+		} elseif (Director::isDev()) {
```
Comment (dhensby) [inline view/Requirements.php]: If we're throwing exceptions, they should always be thrown, not just in some cases

Comment (sminnee) [inline view/Requirements.php]: As highlighted in #6373, this breaks behaviour that modules/sites were relying on: by silently ignoring missing files, modules used this to optionally include a file if a theme provided. So this is a semver failure.

**Pull request merged**